### PR TITLE
Add front-end media lib usage events

### PR DESCRIPTION
**Pull request by soupette**
Signed-off-by: soupette <cyril.lpz@gmail.com>

#### Description of what you did:
This PR adds event in order to improve the usage of Strapi

### CODE REVIEW

File: packages/strapi-plugin-upload/admin/src/components/EditForm/index.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ const EditForm = forwardRef(
       try {
         const file = await getCroppedResult();
 
-        onSubmitEdit(e, shouldDuplicate, file);
+        onSubmitEdit(e, shouldDuplicate, file, true);
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-upload/admin/src/components/EditForm/index.js:175]: I'm not sure but why is it always true? It should be the same value as the `shouldDuplicate` parameter?

File: packages/strapi-plugin-upload/admin/src/containers/ModalStepper/index.js [hunk 2]
```diff
@@ -255,6 +262,9 @@ const ModalStepper = ({
   };
 
   const handleSetCropResult = blob => {
+    // Emit event : the user cropped a file that is not uploaded
+    emitEvent('didCropFile', { duplicatedFile: null, location: 'upload' });
```
Comment (Aurelsicoko) [inline packages/strapi-plugin-upload/admin/src/containers/ModalStepper/index.js:266]: I post the comment here, but the `did` event should be after the dispatch to ensure the job has been done. I might be wrong just asking :)

File: packages/strapi-plugin-upload/admin/src/components/EditForm/index.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/components/EditForm/index.js:175]: The fourth argument is used to know if the user cropped a file. As we use the same function to update the file info I need more information to emit the correct event.

File: packages/strapi-plugin-upload/admin/src/containers/ModalStepper/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/containers/ModalStepper/index.js:266]: Well we cannot have an error here as we are just setting a new file in our reducer.

**Pull request merged**